components/editor/modules/embed/index.js

```diff
@@ -0,0 +1,210 @@
+import React from 'react'
+import { matchBlock } from '../../utils'
+import { findOrCreate } from '../../utils/serialization'
+import MarkdownSerializer from 'slate-mdast-serializer'
+import Loader from '../../../Loader'
+import { colors } from '@project-r/styleguide'
+import { css } from 'glamor'
+import { gql, graphql } from 'react-apollo'
+
+import embedFromUrlPlugin from './embedFromUrlPlugin'
+
+const getEmbed = gql`
+query getEmbed($url: String!) {
+  embed(url: $url) {
+    __typename
+    ... on EmbedInterface {
+      id
+    }
+    ... on Twitter {
+      text
+      userId
+      userName
+      userScreenName
+    }
+    ... on Youtube {
+      userId
+      userName
+      thumbnail
+    }
+    ... on Vimeo {
+      userId
+      userName
+      thumbnail
+    }
+  }
+}
+`
+
+const EmbedLoader =
+  Component =>
+  ({
+    loading,
+    error,
+    embed,
+    data,
+    ...props
+  }) => (
+    <Loader loading={loading} error={error} render={() => {
+      const { node, editor } = props
+      const active = editor.value.blocks.some(
+        block => block.key === node.key
+      )
+      return (
+        <div
+          {...styles.border}
+          data-active={active}
+          contentEditable={false}
+        >
+          <Component
+            embed={data.embed}
+            {...props}
+          />
+        </div>
+      )
+    }} />
+  )
+
+const connect = graphql(getEmbed, {
```

Comment:
use `withApollo` and run the query once in `componentDidMount` then write the whole response to slate state and later mdast

Reply:
```js
this.props.client
  .query({
    query: getEmbed,
    variables: {url}
  })
  .then(({ data }) => {
    this.props.editor.change((change) => {
      change.setNodeByKey(this.props.node.key, {data})
    })
  })
```

Reply:
Will gladly do that, once we've settled with a set of values we want to have persisted to Github. Has design dependencies.

Reply:
As a matter of fact, this was the way I had it implemented at first. @patte's input was, that we don't know yet, how up to date e.g. Twitter share info should be or how to handle mutated embedded data (e.g. deleted videos). 
Again, I'd love to get rid of the Apollo injection within the render component, but I'd like to have a little more consensus on what embeds can and can't do before implementing another draft, that will most likely not satisfy everybody out of the box.

Reply:
I'm pretty sure that it's the right decision to only fetch once in the editor. If we later decide to add e.g. live tweet data we would still only fetch once the static meta data (id, tweet text & author etc.) and then write a front-end component that can fetch additional live data e.g. after mount client side. A bonus feature would be to add a property form with a refetch button, however that is totally bonus because one can also simply re-create the embed by pasting the url again. Refetching unconsciously will destroy the benefit of fetch data and saving it on our side (e.g. deleted tweet).
@patte agree?

Reply:
I totally agree to fetch all required information (strip TW/FB API response, to what we display) in the editor and persist it into mdast, so that rendering an embed works without hitting any API. During brainstorming I thought that it would be very nice to request live data (retweet-, like-, response-count, if the embed was deleted meanwhile) at mount client side from the API and show it in the embed. If that's too much trouble, let it be.
To be clear live updating meta data (must) never undermine the benefit of caching the data on our side, this data is just augmented with what happened meanwhile. So if e.g. an embedded content gets deleted we would show that in the embed, but to remove the embed we would remove it from the article and republish it.